catalog/src/main/java/ch/unibas/dmi/dbis/polyphenydb/catalog/CatalogImpl.java

```diff
@@ -541,54 +542,78 @@ public void setPrimaryKey( long tableId, Long keyId ) throws GenericCatalogExcep
 
 
     /**
-     * Adds a placement for a table
+     * Adds a placement for a column.
      *
      * @param storeId The store on which the table should be placed on
-     * @param tableId The id of the table to be placed
+     * @param columnId The id of the column to be placed
      * @param placementType The type of placement
+     * @param physicalSchemaName The schema name on the data store
+     * @param physicalTableName The table name on the data store
+     * @param physicalColumnName The column name on the data store
      * @throws GenericCatalogException A generic catalog exception
      */
     @Override
-    public void addDataPlacement( int storeId, long tableId, PlacementType placementType ) throws GenericCatalogException {
+    public void addColumnPlacement( int storeId, long columnId, PlacementType placementType, String physicalSchemaName, String physicalTableName, String physicalColumnName ) throws GenericCatalogException {
         try {
             val transactionHandler = XATransactionHandler.getOrCreateTransactionHandler( xid );
             CatalogStore store = Statements.getStore( transactionHandler, storeId );
-            CatalogTable table = Statements.getTable( transactionHandler, tableId );
-            Statements.addDataPlacement( transactionHandler, store.id, table.id, placementType );
-        } catch ( CatalogConnectionException | CatalogTransactionException | UnknownStoreException | UnknownTableTypeException | UnknownTableException e ) {
+            CatalogColumn column = Statements.getColumn( transactionHandler, columnId );
+            CatalogTable table = Statements.getTable( transactionHandler, column.tableId );
+            Statements.addColumnPlacement( transactionHandler, store.id, column.id, table.id, placementType, physicalSchemaName, physicalTableName, physicalColumnName );
```

Comment:
Fetching the `CatalogTable` isn't required, as we only use `table.id`, which should be the same as `column.tableId`.

```suggestion
            Statements.addColumnPlacement( transactionHandler, store.id, column.id, column.tableId, placementType, physicalSchemaName, physicalTableName, physicalColumnName );
```